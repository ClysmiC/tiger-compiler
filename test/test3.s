.data

#User-created variables

#Compiler-created variables.
_return_it_arg0: .word 0
_return_it_n_int: .word 0
_i0: .word 0
__return_it_arg0: .word 0


.text

return_it:
move $t1, $t0

program_start:
move $t1, $t0
move $t1, $t0
