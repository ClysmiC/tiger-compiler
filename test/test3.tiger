let

        function return_it(n:int) : int begin
                return(n);
        end;
in
        return_it(5);
end
