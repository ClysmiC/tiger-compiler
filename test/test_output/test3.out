LET
FUNCTION
ID
LPAREN
ID
COLON
INT
RPAREN
COLON
INT
BEGIN
RETURN
LPAREN
ID
RPAREN
SEMI
END
SEMI
IN
ID
LPAREN
INTLIT
RPAREN
SEMI
END
EOF

Successful parse